x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_flyout/form_components/key_engine_access.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  EuiFormRow,
+  EuiRadio,
+  EuiCheckbox,
+  EuiText,
+  EuiTitle,
+  EuiSpacer,
+  EuiPanel,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { AppLogic } from '../../../../app_logic';
+import { CredentialsLogic } from '../../credentials_logic';
+
+export const FormKeyEngineAccess: React.FC = () => {
+  const { myRole } = useValues(AppLogic);
+  const { setAccessAllEngines } = useActions(CredentialsLogic);
+  const { fullEngineAccessChecked } = useValues(CredentialsLogic);
+
+  return (
+    <>
+      <EuiSpacer />
+      <EuiFormRow
+        label={i18n.translate('xpack.enterpriseSearch.appSearch.tokens.formEngineAccess.label', {
+          defaultMessage: 'Engine Access Control',
+        })}
+      >
+        <>
+          <EuiRadio
+            name="engine_access"
+            id="all_engines"
+            label={
+              <>
+                <EuiTitle size="xxs">
+                  <h3>
+                    {i18n.translate(
+                      'xpack.enterpriseSearch.appSearch.tokens.formEngineAccess.fullAccess.label',
+                      { defaultMessage: 'Full Engine Access' }
+                    )}
+                  </h3>
+                </EuiTitle>
+                <EuiText size="s">
+                  {i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.tokens.formEngineAccess.fullAccess.helpText',
+                    { defaultMessage: 'Access to all current and future Engines.' }
+                  )}
+                </EuiText>
+              </>
+            }
+            hidden={!myRole.canAccessAllEngines}
+            checked={fullEngineAccessChecked}
+            value={fullEngineAccessChecked.toString()}
+            onChange={() => setAccessAllEngines(true)}
+          />
+          <EuiSpacer size="s" />
+          <EuiRadio
+            name="engine_access"
+            id="specific_engines"
+            label={
+              <>
+                <EuiTitle size="xxs">
+                  <h3>
+                    {i18n.translate(
+                      'xpack.enterpriseSearch.appSearch.tokens.formEngineAccess.limitedAccess.label',
+                      { defaultMessage: 'Limited Engine Access' }
+                    )}
+                  </h3>
+                </EuiTitle>
+                <EuiText size="s">
+                  {i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.tokens.formEngineAccess.limitedAccess.helpText',
+                    { defaultMessage: 'Limit key access to specific Engines.' }
+                  )}
+                </EuiText>
+              </>
+            }
+            checked={!fullEngineAccessChecked}
+            value={(!fullEngineAccessChecked).toString()}
+            onChange={() => setAccessAllEngines(false)}
+          />
+        </>
+      </EuiFormRow>
+      {!fullEngineAccessChecked && <EngineSelection />}
+    </>
+  );
+};
+
+export const EngineSelection: React.FC = () => {
```

Comment:
Definitely my opinion, but I don't think there's a standard/best practice for this. I generally prefer taking advantage of multiple exports (the same way we export multiple string consts, utility fns, etc.) and I don't 100% see why components are significantly different.

In this case EngineSelection is a subcomponent and primarily exported to make unit testing easier - but I think that's a feature, not a bug, of unit testing, because it helps us break code down into substantially more readable blocks. I also don't think exporting for the sake of a test file is a major faux pas or anything - conceptually it's not any different from mocks or other const's exported to make tests easier.